src/bus/communication_bus.cpp

```diff
@@ -0,0 +1,45 @@
+#include <map>
+#include <string>
+#include <mutex>  // NOLINT [build/c++11]
+
+#include "bus/communication_bus.hpp"
+
+namespace mesos {
+namespace serenity {
+
+/** Static functions */
+
+Option<process::UPID> CommunicationBus::getBusUPID(
+    const std::string& _busName) {
+  std::lock_guard<std::mutex> lock(busMapLock);
+  return CommunicationBus::_getBusUPID(_busName);
+}
+
+
+Option<process::UPID> CommunicationBus::_getBusUPID(
```

Comment:
This looks more like an internal function rather than a continuation. Could we unify them and or, guard the bus in a class that owns the lock? It is easy to forget to acquire the lock before accessing the map